diff --git a/node_modules/@safe-global/safe-core-sdk-utils/dist/src/eip-3770/config.js b/node_modules/@safe-global/safe-core-sdk-utils/dist/src/eip-3770/config.js
index fe3c79b..758693e 100644
--- a/node_modules/@safe-global/safe-core-sdk-utils/dist/src/eip-3770/config.js
+++ b/node_modules/@safe-global/safe-core-sdk-utils/dist/src/eip-3770/config.js
@@ -13,6 +13,8 @@ exports.networks = [
     { chainId: 18, shortName: 'tst' },
     { chainId: 25, shortName: 'cro' },
     { chainId: 28, shortName: 'bobarinkeby' },
+    { chainId: 30, shortName: 'rsk' },
+    { chainId: 31, shortName: 'trsk' },
     { chainId: 39, shortName: 'u2u' },
     { chainId: 40, shortName: 'telosevm' },
     { chainId: 41, shortName: 'telosevmtestnet' },
